Sources/ApolloCodegenLib/Templates/OperationIdentifiersTemplate.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+import ApolloUtils
+
+/// Provides the format to output a file used for APQ registration.
+struct OperationIdentifiersTemplate {
+  /// Collection of operation identifiers to be serialized.
+  let operationIdentifiers: [OperationIdentifier]
+
+  var template: TemplateString {
+    TemplateString(
+    """
+    {
+      \(operationIdentifiers.map({ operation in
+      """
+      "\(operation.hash)": {
+        "name": "\(operation.name)",
+        "source": "\(operation.source.jsonEscaped())"
+      }
+      """}), separator: ",\n")
+    }
+    """
+    )
+  }
+
+  func render() -> String {
+    template.description
+  }
+}
+
+fileprivate extension String {
+  func jsonEscaped() -> String {
```

Comment:
I'm not sure of another 'standard' way to do this.

Reply:
I'm second-guessing myself now - maybe it would be better to just implement custom `Codable` conformance and know that JSON escaping/formatting will be 100% correct?

Reply:
I tried `Codable` yesterday but a plain `Dictionary` didn't preserve order which made testing fail randomly and `OrderedDictionary` serialized with array opening/closing braces and the object format wasn't right.

I'll try this approach but I suspect the lack of ordering could still cause an issue with tests.